products/compute/api.yaml

```diff
@@ -5841,6 +5841,170 @@ objects:
         description: |
           URL of the disk type resource describing which disk type to use to
           create the disk. Provide this when creating the disk.
+  - !ruby/object:Api::Resource
+    name: 'ResourcePolicy'
+    min_version: beta
+    kind: 'compute#resourcePolicy'
+    base_url: projects/{{project}}/regions/{{region}}/resourcePolicies
+    input: true
+    has_self_link: true
+    description: |
+      A policy that can be attached to a resource to specify or schedule actions on that resource.
+    async: !ruby/object:Api::Async
+      operation: !ruby/object:Api::Async::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/global/operations/{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::Async::Result
+        path: 'targetLink'
+      status: !ruby/object:Api::Async::Status
+        path: 'status'
+        complete: 'DONE'
+        allowed:
+          - 'PENDING'
+          - 'RUNNING'
+          - 'DONE'
+      error: !ruby/object:Api::Async::Error
+        path: 'error/errors'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: region
+        resource: Region
+        imports: name
+        description: Region where resource policy resides.
+        input: true
+        required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        required: true
+        description: |
+          The name of the resource, provided by the client when initially creating
+          the resource. The resource name must be 1-63 characters long, and comply
+          with RFC1035. Specifically, the name must be 1-63 characters long and
+          match the regular expression [a-z]([-a-z0-9]*[a-z0-9])? which means the
+          first character must be a lowercase letter, and all following characters
+          must be a dash, lowercase letter, or digit, except the last character,
+          which cannot be a dash.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'snapshotSchedulePolicy'
+        description: |
+          Policy for creating snapshots of persistent disks.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'schedule'
+            description: |
+              Contains one of an `hourlySchedule`, `dailySchedule`, or `weeklySchedule`.
```

Comment:
does that mean only one can be given? if so, probably should add conflicts_with for schedules

Reply:
talked offline and conflicts_with currently doesn't work for nested fields, leaving out for now (adding todo)

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/issues/1881